iceoryx_dust/include/iceoryx_dust/posix_wrapper/named_pipe.hpp

```diff
@@ -122,6 +114,10 @@ class NamedPipe : public DesignPattern::Creation<NamedPipe, IpcChannelError>
     template <typename Prefix>
     static IpcChannelName_t convertName(const Prefix& p, const IpcChannelName_t& name) noexcept;
 
+    /// @brief destroys an initialized named pipe.
+    /// @return is always successful
```

Comment:
Yes you are completely right. But our ugly design by abstracting the ipc channel with a type alias forced us to implement non-sense methods with useless return values to the named pipe and the unix domain socket.

This has to be refactored but first we need to introduce a class which unifies the different behavior of those ipc channels. This is what this PR is doing by using the `IpcInterface` for exactly this.

A follow up PR would refactor the unix domain socket, the named pipe and the message queue so that they use the builder pattern and the interface is adjusted so that it actually makes sense.
The `IpcInterface` then takes care of it so that this class behaves always in the same fashion independent of the underlying ipc channel.

Reply:
Ok, understood, thanks. Could you please add this as a todo to the issue?